x-pack/plugins/timelines/public/components/actions/timeline/cases/add_to_case_action.tsx

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import React, { memo, useState, useCallback, useMemo } from 'react';
+import {
+  EuiPopover,
+  EuiButtonIcon,
+  EuiContextMenuPanel,
+  EuiText,
+  EuiContextMenuItem,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import { Case, CaseStatuses, StatusAll } from '../../../../../../cases/common';
+import { Ecs } from '../../../../../common/ecs';
+import { useGetUserCasesPermissions } from '../../../../hooks/use_get_user_cases_permissions';
+import { useKibana } from '../../../../../../../../src/plugins/kibana_react/public';
+import { TimelinesStartServices } from '../../../../types';
+import { ActionIconItem } from '../../action_icon_item';
+import { CreateCaseFlyout } from './create/flyout';
+import { createUpdateSuccessToaster } from './helpers';
+import * as i18n from './translations';
+
+export interface AddToCaseActionProps {
+  ariaLabel?: string;
+  ecsRowData: Ecs;
+  appId: string;
+}
+// FIXME: DEDUPE
+export const APP_ID = 'securitySolution';
+export enum SecurityPageName {
+  overview = 'overview',
+  detections = 'detections',
+  alerts = 'alerts',
+  rules = 'rules',
+  exceptions = 'exceptions',
+  hosts = 'hosts',
+  network = 'network',
+  timelines = 'timelines',
+  case = 'case',
+  administration = 'administration',
+  endpoints = 'endpoints',
+  policies = 'policies',
+  trustedApps = 'trusted_apps',
+  eventFilters = 'event_filters',
+}
+interface UseControlsReturn {
+  isControlOpen: boolean;
+  openControl: () => void;
+  closeControl: () => void;
+}
+
+export const appendSearch = (search?: string) =>
+  isEmpty(search) ? '' : `${search?.startsWith('?') ? search : `?${search}`}`;
+
+export const getCaseUrl = (search?: string | null) => `${appendSearch(search ?? undefined)}`;
+
+export const getCreateCaseUrl = (search?: string | null) =>
+  `/create${appendSearch(search ?? undefined)}`;
+
+export const getCaseDetailsUrl = ({
+  id,
+  search,
+  subCaseId,
+}: {
+  id: string;
+  search?: string | null;
+  subCaseId?: string;
+}) => {
+  if (subCaseId) {
+    return `/${encodeURIComponent(id)}/sub-cases/${encodeURIComponent(subCaseId)}${appendSearch(
+      search ?? undefined
+    )}`;
+  }
+  return `/${encodeURIComponent(id)}${appendSearch(search ?? undefined)}`;
+};
+
+// FIXME DEDUPE END
+
+interface PostCommentArg {
+  caseId: string;
+  data: {
+    type: 'alert';
+    alertId: string | string[];
+    index: string | string[];
+    rule: { id: string | null; name: string | null };
+    owner: string;
+  };
+  updateCase?: (newCase: Case) => void;
+  subCaseId?: string;
+}
+
+const AddToCaseActionComponent: React.FC<AddToCaseActionProps> = ({
+  ariaLabel = i18n.ACTION_ADD_TO_CASE_ARIA_LABEL,
+  ecsRowData,
+  appId,
```

Comment:
instead of `appId` call it `featureId` so everything start to make sense with the above comment

Reply:
leaving appId to be used with navigation functions